examples/react-navigation/README.md

```diff
@@ -1,9 +1,16 @@
 # RNTL example app for React Navigation
 
-This example shows how to write integration tests using React Navigation without mocking it.
+This example shows how to write integration tests using React Navigation without mocking it. Presented approach has been consulted with and influenced by React Navigation team.
 
-There are two types of tests:
-1. integration tests operating on whole navigators, they should use `renderNavigator` helper to render a navigator component used in the app. It is useful when you want to test a scenario that includes multiple screens.
-2. single screen tests where you would pass mock `navigation` prop, built using `buildNavigationMock()` helper, and `route` prop to the screen component using regular `render` function.
+
+## Recommended tests
+
+There are two types of recommeded tests:
+1. Tests operating on navigator level - these use `renderNavigator` helper to render a navigator component used in the app. It is useful when you want to test a scenario that includes multiple screens.
+2. Tests operating on single screen level - these use regular `render` helper but require refactoring screen components into `Screen` and `ScreenContent` components. Where `Screen` receives React Navigation props and/or uses hooks like `useNavigation` while `ScreenContent` does not have a direct relation to React Navigation API but gets props from `Screen` and calls relevant callbacks to trigger navigation.
+
+## Non-recommended tests
+
+There also exists another popular type of screen level tests, where users mock React Navigation objects like `navigation`, `route` and/or hooks like  `useNavigation`, etc. This type is not recommended by the React Navigation team as it leads to fragile tests that can lead to false positive and false negative result. While the recommended approach is arguable harder it should lead to more robust tests helping to properly identify potential issues in test. 
```

Comment:
Consider rephrasing like this. I'm linking to TL's guiding principles and also not mentioning that our testing setup is "arguably harder", it's counter productive
```suggestion
There also exists another popular type of screen level tests, where users mock React Navigation objects like `navigation`, `route` and/or hooks like  `useNavigation`, etc. We don't recommend this way of testing. **Mocking internal parts of the libraries is effectively testing implementation details, which goes against the Testing Library's [Guiding Principles](https://testing-library.com/docs/guiding-principles/)**.
```